src/map/packet_system.cpp

```diff
@@ -2815,6 +2816,7 @@ void SmallPacket0x05C(map_session_data_t* session, CCharEntity* PChar, CBasicPac
             updatePosition = luautils::OnEventFinish(PChar, EventID, Result) == 1;
             if (PChar->m_event.EventID == EventID)
             {
+                PChar->m_Substate = CHAR_SUBSTATE::SUBSTATE_NONE;
```

Comment:
Will need one of these in server-outgoing [0x52 (NPC Release)](https://github.com/project-topaz/topaz/blob/release/src/map/packets/release.cpp) too, for event skips (hit during CS) and the !release command.

There's another packet to potentially look into too, but I'll hit you up on Discord about it.